### DISCUSSION TITLE
Hallo Jan ...

### DISCUSSION THREAD

**Discussion by Joerki** [jkowalleck/jkowalleck#5] [HELLO]
Hallo Jan, 
Gerade eben war ich das erste Mal auf deinem Profil und melde mich mal auf diesem Weg.
Ich habe gesehen, dass du in der CycloneDX Community tief verankert bist.
Seit ein paar Jahren habe u.a. ich Aufgaben in unserer Firma um unsere OSS Richtlinie umzusetzen, d.h. SBOMs in CycloneDX 1.4  zu erstellen und Tools zu entwickeln, die das unterstützen.
Einen Großteil meiner Arbeit verbringe ich auch damit, nicht standardkonforme Lizenzangaben von Autoren zu untersuchen. Diese Quell-SBOMs sind in der Regel keine CycloneDX-SBoms, und ich konvertiere sie in CycloneDX 1.4 mit dem Inhalt, der in unserer SBom-Definition festgelegt ist. Dazu gehören auch eigene Properties, die wir gemäß der Taxonomie definieren und ggf. auch veröffentlichen wollen.
Ich habe mit völlig verschiedenen Ökosystemen zu tun wie Python inkl. Pyinstaller, Npm, (Windows) Dotnet, ansatzweise auch mit Zephyr und Yocto sowie Docker Containern, in denen wir Debian als Laufzeitumgebung für Applikationen/Services einsetzen, die in der Cloud oder auf Geräten (ARM64) und PCs laufen. Außerdem haben wir Projekte, bei denen wir keine Paketmanager einsetzen, sondern die Komponenten manuell erfassen müssen.
Wir haben auch schon verschiedene kommerzielle Lösungen evaluiert, ohne jedoch eine für alle zufriedenstellende 
Lösung zu finden. Ich gehöre zu denjenigen im Unternehmen, die nicht nur im Silo ihrer eigenen Ökosysteme denken und arbeiten, sondern einen möglichst vereinheitlichten Workflow und ein weitestgehend gemeinsames Tooling anstreben.
Das Thema Vulnerabilities sehen wir in diesem Kontext natürlich auch, das ist aber nicht so kompliziert. 
Soweit möglich habe ich selbst Syft von Anchore (auch hier auf Github) mit Syft-Json als meinen Input und Grype für die Vuln-Analyse eingesetzt. Es werden aber auch andere Tools eingesetzt, die nach meiner Sichtung bzgl. Licensing starke Defizite aufweisen.
Ich setze Projekte von nexB/aboutcode-org ein (scancode-licensedb, debian-inspector, license-expression, experimentell scancode-toolkit, dejacode), die hier auf Github zu finden sind.

Ich hoffe, dieser Kontext zu mir ist hilfreich für die Issues, die wir diskutieren.
Bin auch generell an einer Vernetzung interessiert, kann aber Dinge wie Slack nur privat einsetzen.
Dies ist auch mein privater Account.

Viele Grüße 
Jörg 
("Joerki")

**Comment by jkowalleck** [comment]
Danke für die aufschlussreiche Nachricht, @Joerki.
Ich bin grade kurz angebunden, ich melde mich, sobald es geht.
Fuege mich gern auf linked-in oder anderen Netzwerken hinzu.

**Comment by Joerki** [comment]
Übrigens ein Missverständnis: Mein Problem mit der Pythonkomponente ist nicht gelöst. Ich kann sie so nicht einsetzen und werde wohl auf meine eigene Implementierung ausweichen.
Ich weiß, dass ein Schema nur eine Abbildung einer Spezifikation ist. Aber wenn ich cyclonedx.org besuche und auf documentation gehe, dann lande ich doch bei der aus dem  JSON-Schema generierten Dokumentation.
Früher habe ich über 15 Jahre im Mobilfunksektor gearbeitet und damit mit vielen 3GPP Spezifikationen zu tun gehabt. Nicht nur als Entwickler, sondern auch als Tester und QA Lead musste ich mich mit Specs und Implementierungen umgehen.

**Reply by jkowalleck** [reply]
verstanden.
wenns nicht geloest ist schrieb nen kommentar drunter, dann mach ich den wieder auf.

**Comment by Joerki** [comment]
Hallo Jan, 
ich möchte meinen lokal erstellten und bearbeiteten Branch zu #765 gerne pushen, aber dafür müsste das Issue erst einmal mir zugewiesen werden und ich auch authorisiert werden. Da dies mein erster Commit ist, kenne ich mich nicht aus.
Kannst du mir helfen?
Alle Punkte aus der CONTRIBUTING.md habe ich ausgeführt und der poetry run tox run ist zufrieden.
Gruß,
Jörg

**Reply by jkowalleck** [reply]
Moin Jörg,

du hast also einen lokalen git-Branch?
Dann pushe den doch in einen Fork des Ziel-Repositorien, und pull-requeste dann deiene Branch aus dem remote/origin Fork in das remote/upstream original.

- see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo
- see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork 

Hilft dir das? 


Hintergrund: fork/pull-request erlaubt es jedem beim Projekt beizutragen. Dafuer sind keine Berechtigungen notwendig und contributor koenen asynchron in ihrem tempo mitmachen.

**Comment by jkowalleck** [comment]
re:  https://github.com/jkowalleck/jkowalleck/discussions/5#discussioncomment-12092266
>  Kannst du mir helfen?
> Alle Punkte aus der CONTRIBUTING.md habe ich ausgeführt und der poetry run tox run ist zufrieden.

waere es deiner meinung nach sinnvoll, das vorgehen mit fork/pullrequest in der `CONTRIBUTING.md` zu beschreiben? 
Falls ja, pullrequeste doch gerne eine anleitung 👍

**Reply by Joerki** [reply]
OK, ich überlege mir einen Text für meinen nächsten Commit im Branch

https://github.com/Joerki/cyclonedx-python-lib

Würdest du einmal in meinen Commit 

https://github.com/CycloneDX/cyclonedx-python-lib/compare/main...Joerki:cyclonedx-python-lib:fix/issue765

schauen bevor ich den PR mache? Ich habe die Abhängigkeit zur SPDX-Datei in _res ausgebaut, da die SPDX Liste aus den Daten von license-expression extrahierbar ist. Die SPDX Liste in python-license-lib muss nicht mehr gepflegt werden und beim Bauen von Programmen muss man nur noch darauf achten, auch die aktuelle license-expression Bibliothek zu haben (das war ja auch vorher schon notwendig). Sollte ich einen wichtigen Aspekt der Präsenz der Referenzierung in spdx.py/test_spdx.py nicht kennen, gib mir bitte Bescheid.

Ansonsten stelle ich den PR.

Gruß,
Jörg

**Reply by jkowalleck** [reply]
> Würdest du einmal in meinen Commit
> 
https://github.com/CycloneDX/cyclonedx-python-lib/compare/main...Joerki:cyclonedx-python-lib:fix/issue765
> schauen bevor ich den PR mache?

Genau dafür ist der pull-request(PR) review. da kann ich den diff untersuchen, und annotieren. 
Eroeffne den PR und ich gucks mir an. :-)
